### ISSUE TITLE
True ANON User Mode

### ISSUE THREAD

**Issue by starlilyth** [grafana/grafana#12019]
Because Grafana is the #1 TSD graph templating solution available I would like to request an actual, useable ANONYMOUS user mode, one that does not indicate or allow any editing of the exported realtime graphs. 

I am providing graphs via the "Embed" feature in an iframe to a web application with many users. These graphs are exported by enabling 'anonymous' access, but they are in not in any way something I want to put in the hands of 'anonymous' users in production: 

I can use "&inactive" to disable the drop menu on a graph that has been exported, but the "+" grab handle or finger pointer is still displayed when mousing over the header. 
Even with "&inactive", Anon users can still edit legend details (axis, color) by clicking on the items color. 
Editing by Anon users is possible if the dashboard does not have "Editable" unchecked. 

Bottom line: ANONYMOUS users should NEVER have edit access to any part of the exported graph, or indicators to edit options, regardless of settings on the dashboard or flags used in the url! This seems an obvious and common sense safeguard; I am surprised it does not work this way already, hence my request. 

I understand that this may not have been something considered in your original roadmap or current set of milestones, but you are doing the project an enormous disservice by not recognizing the potential application of this tool in the absence of any better solutions. I would be grateful for your serious consideration of this request. 

Grafana v5.1.3
InfluxDB 1.5.2
Centos 7

Thank you, 
Lily

**Comment by torkelo** [comment]
We could disable the panel menu in solo/embed mode. Would not add any true security but would be less confusing for the user interacting with the iframed panel.

**Comment by starlilyth** [comment]
Please also disable the legend edit menu in the same mode, as this poses the same confusion problem. 

I appreciate your attention to this issue.

**Comment by dastrix80** [comment]
Is there any update to this request? I too would like all the menus etc removed from an iframe embedded graph

**Comment by rtandon98** [comment]
+1

**Comment by z11aalex** [comment]
+1 
this feature shall have a greater attention from grafana community

**Comment by balabalazhoucj** [comment]
+1

**Comment by daFritz84** [comment]
+1

**Comment by SomeFixItDude** [comment]
+1
Makes embedding much more usable.

**Comment by MRakeshHarsha** [comment]
+1
any update on this?

**Comment by sol-nikita** [comment]
+1

**Comment by hanynowsky** [comment]
This issue makes the snapshoting useless.
adding:  &inactive=1  to the URL does not help much.

**Comment by facetoe** [comment]
+1

**Comment by mengon05** [comment]
I made a workaround for the Embed charts, i just put a div on top in the place where the menu is. You can use the same pattern to create a transparent div as a "shield" for the chart legends in order to avoid user interaction

```
<div class="chart-container"> 
    <div></div>
    <iframe src="the/grafana/url" frameborder="0"></iframe>
</div>
```
```
.chart-container{
    position: relative;
    div:first-child {
        background: white;
        height: 1.8em;
        width: 100%;
        position: absolute;
    }
    iframe{
        width:100%;
        height: 100%;
    }
}
```

**Comment by tuhao1020** [comment]
any update on this request?

**Comment by ntompson** [comment]
Seriously - this isn't an option?

+1 for me too

**Comment by antonpod** [comment]
+1
Please implement this, I'm using negative margin meanwhile to hide the header and the sidebar.

**Comment by syedsfayaz** [comment]
+1

**Comment by Bots4You-Heimel** [comment]
+1

**Comment by YuxiJin-tobeyjin** [comment]
+1

**Comment by flyboy013** [comment]
+1

**Comment by yogi4yu** [comment]
+1 this is a must have feature

**Comment by yogi4u** [comment]
+1 really a useful feature

**Comment by varunshah1106** [comment]
+1

**Comment by 1300371** [comment]
+1

**Comment by razie** [comment]
how do we vote for this? Is someone counting comments or is there something smarter?

**Comment by fabionaspolini** [comment]
+1

**Comment by Saoni** [comment]
Is there any update on this issue? Can anyone provide the solution to this?

**Comment by tmm1** [comment]
If you can inject CSS into the iframe, something like this works:

```
.panel-header:hover{ background-color: transparent; }
.panel-title-container{ cursor: auto; }
.panel-title-text{ cursor: auto; }
.panel-menu-toggle{ display: none; }
.panel-menu-container{ display: none; }
```

**Comment by yogi4u** [comment]
It also make sense to disable the menu for non-admin/viewer/guest user or based on user role.

**Comment by oliverwreath** [comment]
+1 this feature request deserve more attention from the Grafana community.

**Comment by hanumesh** [comment]
+100 this feature request deserve more attention from the Grafana community.

**Comment by hanumesh** [comment]
https://community.grafana.com/t/disable-view-share-and-more-menu-from-graph/26572/12 

We require more attention from Graphana community

**Comment by faizansalamat** [comment]
A quick and easy hack to hide the graph controls : 

We are going to play with the themes.

Grafana has two Themes " Light " & " Dark ".

I edited my light theme css by hiding all the menus, sidebar , and page titles. This light theme was shared with the end users while at the back I was monitoring all the alert on the dark theme. You can pass Theme in the URL **https://[dashboardurl]&theme=light**

in short " hide all the extra elements from one theme and share it with the users while you use the other theme to play around with the data "

head over to the build folder. ( located under extracted grafana folder )
**- D:\grafana-8.0.4.windows-amd64\grafana-8.0.4\public\build\ 
- Look for grafana.light.d6a5ae6ae74fba833b3d ( if you want to edit the light theme and hide the controls )**
                                                         or
**- grafana.dark.d6a5ae6ae74fba833b3d ( if you want to edit the dark theme and hide controls )**

**here the css which you will paste at the bottom of your css :** 

**.page-toolbar .css-1hik5m6 { display:none !important} //to hide toolbar
.panel-title {display:none !important} //to hide graph titles
.submenu-controls { display: none !important} //to hide sub menu controls
.css-1rf5v84 {display: none !important} //to hide toolbar inner
.css-6v8co2 {display: none !important} //to hide  toolbar inner
.sidemenu {diplay: none !important} // to hide sidebar** 

go head save and refresh grafana. 

cheers.

**Comment by figotee3** [comment]
+1

**Comment by ikaranmali** [comment]
> A quick and easy hack to hide the graph controls :
> 
> We are going to play with the themes.
> 
> Grafana has two Themes " Light " & " Dark ".
> 
> I edited my light theme css by hiding all the menus, sidebar , and page titles. This light theme was shared with the end users while at the back I was monitoring all the alerts on the dark theme. You can pass Theme in the URL **https://[dashboardurl]&theme=light**
> 
> in short " hide all the extra elements from one theme and share it with the users while you use the other theme to play around with the data "
> 
> head over to the build folder. ( located under extracted grafana folder )
> 
> *_- D:\grafana-8.0.4.windows-amd64\grafana-8.0.4\public\build*_ **- Look for grafana.light.d6a5ae6ae74fba833b3d ( if you want to edit the light theme and hide the controls )** or **- grafana.dark.d6a5ae6ae74fba833b3d ( if you want to edit the dark theme and hide controls )**
> 
> **here is the css which you will paste at the bottom of your css file :**
> 
> **.page-toolbar .css-1hik5m6 { display:none !important} //to hide toolbar .panel-title {display:none !important} //to hide graph titles .submenu-controls { display: none !important} //to hide sub menu controls .css-1rf5v84 {display: none !important} //to hide toolbar inner .css-6v8co2 {display: none !important} //to hide toolbar inner .sidemenu {diplay: none !important} // to hide sidebar** **.react-resizable-handle{display:none !Important;} // to remove the drag event on the panels**
> 
> go head save and refresh grafana.
> 
> cheers.

Hi above CSS settings works perfectly fine in v9.3.1 Windows, and now while trying same in linux (under /usr/share/grafana/public/build/) it fails to do so!!

**Comment by MarBlack23** [comment]
Is there any method for version 10? @faizansalamat

**Comment by ikaranmali** [comment]
Version 10 seems to not accepting these changes, on version 9 the trick works fine!
Please suggest how to do it for version 10.

**Comment by julnegre** [comment]
**Please provide an option to activate the menu or not for anonymous user !!! **

**Comment by curtis1757** [comment]
This seems like such a simple request to implement.  Without it,  it makes grafana useles for my purposes.

The request is over 6 years old!!!!

&kiosk does not fix this problem

**Comment by fboerman** [comment]
+1

**Comment by lnjustin** [comment]
+1
